### PR TITLE
DBAAS Kafka Topics Support

### PR DESCRIPTION
**Pull request by dweinshenker**
**What**

- Adds support for CRUD operations on kafka topics, partitions
- Supports advanced configuration for kafka topics

### CODE REVIEW

File: commands/databases.go [hunk 1]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+	CmdBuilder(cmd, RunDatabaseTopicGet, "get <database-id> <topic-name>", "Retrieve the configuration for a given kafka topic", topicGetDetails, Writer, displayerType(&displayers.DatabaseKafkaTopic{}), aliasOpt("g"))
+	CmdBuilder(cmd, RunDatabaseTopicListPartition, "partitions <database-id> <topic-name>", "Retrieve the partitions for a given kafka topic", topicGetPartitionDetails, Writer, aliasOpt("p"))
+	cmdDatabaseTopicDelete := CmdBuilder(cmd, RunDatabaseTopicDelete, "delete <database-id> <topic-name>", "Deletes a kafka topic by topic name", "", Writer, aliasOpt("rm"))
+	AddBoolFlag(cmdDatabaseTopicDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Deletes the kafka topic without a confirmation prompt")
+	cmdDatabaseTopicCreate := CmdBuilder(cmd, RunDatabaseTopicCreate, "create <database-id>", "Creates a topic for a given kafka database",
+		"This command creates a kafka topic for the specified kafka database cluster, giving it the specified name. Example: doctl databases topics create <cluster_uuid> <topic_name>", Writer, aliasOpt("c"))
+	cmdDatabaseTopicUpdate := CmdBuilder(cmd, RunDatabaseTopicUpdate, "update <database-id>", "Updates a topic for a given kafka database",
```
Comment (danaelhe) [inline commands/databases.go:1856]:
```suggestion
	cmdDatabaseTopicUpdate := CmdBuilder(cmd, RunDatabaseTopicUpdate, "update <database-uuid> <topic-name>", "Updates a topic for a given kafka database",
```

File: commands/databases.go [hunk 2]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+	CmdBuilder(cmd, RunDatabaseTopicGet, "get <database-id> <topic-name>", "Retrieve the configuration for a given kafka topic", topicGetDetails, Writer, displayerType(&displayers.DatabaseKafkaTopic{}), aliasOpt("g"))
+	CmdBuilder(cmd, RunDatabaseTopicListPartition, "partitions <database-id> <topic-name>", "Retrieve the partitions for a given kafka topic", topicGetPartitionDetails, Writer, aliasOpt("p"))
+	cmdDatabaseTopicDelete := CmdBuilder(cmd, RunDatabaseTopicDelete, "delete <database-id> <topic-name>", "Deletes a kafka topic by topic name", "", Writer, aliasOpt("rm"))
+	AddBoolFlag(cmdDatabaseTopicDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Deletes the kafka topic without a confirmation prompt")
+	cmdDatabaseTopicCreate := CmdBuilder(cmd, RunDatabaseTopicCreate, "create <database-id>", "Creates a topic for a given kafka database",
+		"This command creates a kafka topic for the specified kafka database cluster, giving it the specified name. Example: doctl databases topics create <cluster_uuid> <topic_name>", Writer, aliasOpt("c"))
```
Comment (danaelhe) [inline commands/databases.go:1855]:
```suggestion
		"This command creates a kafka topic for the specified kafka database cluster, giving it the specified name. Example: doctl databases topics create <database-uuid> <topic-name> --replication_factor 2 --partition_count 4", Writer, aliasOpt("c"))
```

File: commands/databases.go [hunk 3]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+	- The Id - identifier of the topic partition.
+	- The Size - size of the topic partition, in bytes.
+	- The InSyncReplicas - number of brokers that are in sync with the partition leader.
+	- The EarliestOffset - earliest offset read amongst all consumers of the partition.
+	`
+
+	CmdBuilder(cmd, RunDatabaseTopicList, "list <database-id>", "Retrieve a list of topics for a given kafka database", topicListDetails, Writer, displayerType(&displayers.DatabaseKafkaTopics{}), aliasOpt("ls"))
+	CmdBuilder(cmd, RunDatabaseTopicGet, "get <database-id> <topic-name>", "Retrieve the configuration for a given kafka topic", topicGetDetails, Writer, displayerType(&displayers.DatabaseKafkaTopic{}), aliasOpt("g"))
+	CmdBuilder(cmd, RunDatabaseTopicListPartition, "partitions <database-id> <topic-name>", "Retrieve the partitions for a given kafka topic", topicGetPartitionDetails, Writer, aliasOpt("p"))
+	cmdDatabaseTopicDelete := CmdBuilder(cmd, RunDatabaseTopicDelete, "delete <database-id> <topic-name>", "Deletes a kafka topic by topic name", "", Writer, aliasOpt("rm"))
```
Comment (danaelhe) [inline commands/databases.go:1852]:
```suggestion
	cmdDatabaseTopicDelete := CmdBuilder(cmd, RunDatabaseTopicDelete, "delete <database-uuid> <topic-name>", "Deletes a kafka topic by topic name", "", Writer, aliasOpt("rm"))
```

File: commands/databases.go [hunk 4]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+This command lists the following details for each partition of a given topic in a kafka database cluster:
+
+	- The Id - identifier of the topic partition.
+	- The Size - size of the topic partition, in bytes.
+	- The InSyncReplicas - number of brokers that are in sync with the partition leader.
+	- The EarliestOffset - earliest offset read amongst all consumers of the partition.
+	`
+
+	CmdBuilder(cmd, RunDatabaseTopicList, "list <database-id>", "Retrieve a list of topics for a given kafka database", topicListDetails, Writer, displayerType(&displayers.DatabaseKafkaTopics{}), aliasOpt("ls"))
+	CmdBuilder(cmd, RunDatabaseTopicGet, "get <database-id> <topic-name>", "Retrieve the configuration for a given kafka topic", topicGetDetails, Writer, displayerType(&displayers.DatabaseKafkaTopic{}), aliasOpt("g"))
```
Comment (danaelhe) [inline commands/databases.go:1850]:
```suggestion
	CmdBuilder(cmd, RunDatabaseTopicGet, "get <database-uuid> <topic-name>", "Retrieve the configuration for a given kafka topic", topicGetDetails, Writer, displayerType(&displayers.DatabaseKafkaTopic{}), aliasOpt("g"))
```

File: commands/databases.go [hunk 5]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+	topicGetPartitionDetails := `
+This command lists the following details for each partition of a given topic in a kafka database cluster:
+
+	- The Id - identifier of the topic partition.
+	- The Size - size of the topic partition, in bytes.
+	- The InSyncReplicas - number of brokers that are in sync with the partition leader.
+	- The EarliestOffset - earliest offset read amongst all consumers of the partition.
+	`
+
+	CmdBuilder(cmd, RunDatabaseTopicList, "list <database-id>", "Retrieve a list of topics for a given kafka database", topicListDetails, Writer, displayerType(&displayers.DatabaseKafkaTopics{}), aliasOpt("ls"))
```
Comment (danaelhe) [inline commands/databases.go:1849]:
```suggestion
	CmdBuilder(cmd, RunDatabaseTopicList, "list <database-uuid>", "Retrieve a list of topics for a given kafka database", topicListDetails, Writer, displayerType(&displayers.DatabaseKafkaTopics{}), aliasOpt("ls"))
```

File: commands/databases.go [hunk 6]
```diff
@@ -1554,6 +1556,355 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
...
+	CmdBuilder(cmd, RunDatabaseTopicListPartition, "partitions <database-id> <topic-name>", "Retrieve the partitions for a given kafka topic", topicGetPartitionDetails, Writer, aliasOpt("p"))
+	cmdDatabaseTopicDelete := CmdBuilder(cmd, RunDatabaseTopicDelete, "delete <database-id> <topic-name>", "Deletes a kafka topic by topic name", "", Writer, aliasOpt("rm"))
+	AddBoolFlag(cmdDatabaseTopicDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Deletes the kafka topic without a confirmation prompt")
+	cmdDatabaseTopicCreate := CmdBuilder(cmd, RunDatabaseTopicCreate, "create <database-id>", "Creates a topic for a given kafka database",
+		"This command creates a kafka topic for the specified kafka database cluster, giving it the specified name. Example: doctl databases topics create <cluster_uuid> <topic_name>", Writer, aliasOpt("c"))
+	cmdDatabaseTopicUpdate := CmdBuilder(cmd, RunDatabaseTopicUpdate, "update <database-id>", "Updates a topic for a given kafka database",
+		"This command updates a kafka topic for the specified kafka database cluster. Example: doctl databases topics update <cluster_uuid> <topic_name>", Writer, aliasOpt("u"))
```
Comment (danaelhe) [inline commands/databases.go:1857]:
```suggestion
		"This command updates a kafka topic for the specified kafka database cluster. Example: doctl databases topics update <database-uuid> <topic-name>", Writer, aliasOpt("u"))
```

**Review by danaelhe — CHANGES_REQUESTED** [review]
Looks great! Just some minor nitpicks on documentation.

**Comment by dweinshenker** [comment]
Thanks for the thorough review @danaelhe! agreed with all your suggestions for documentation and made those commits.

**Review by danaelhe — APPROVED** [review]
Looks great! Thank you so much 🚢

**Pull request merged**